activesupport/lib/active_support/core_ext/object/json.rb

```diff
@@ -46,7 +46,7 @@ def to_json(options = nil)
 end
 
 [Enumerable, Object, Array, FalseClass, Float, Hash, Integer, NilClass, String, TrueClass].reverse_each do |klass|
-  klass.prepend(ActiveSupport::ToJsonWithActiveSupportEncoder)
+  klass.include(ActiveSupport::ToJsonWithActiveSupportEncoder)
```

Comment:
Are you sure we can call `include` here? We need to call `super` in the `to_json` implementation

Reply:
We're guaranteed to load after JSON because of the `require "json"` on [line 4](https://github.com/rails/rails/blob/23238230818c1b1cf8cae39d77a50c9755655396/activesupport/lib/active_support/core_ext/object/json.rb#L4). While there could be something interfering with the typical load process, the test suite passes and I couldn't find a failure case locally. I tried with a few different data types, including a custom `Struct` so I'd hit the `Object` case, and the results were the same with either `prepend` or `include`.

If the `reverse_each` is changed to `each` then there's a behavioral difference between `prepend` (tests pass in both directions) and `include` (tests only pass with `reverse_each`), but the `reverse_each` is already there.